### DISCUSSION TITLE
Mapbox's new globe view cannot work with _GlobeView

### DISCUSSION THREAD

**Discussion by rajatmann100** [visgl/deck.gl#7118] [Q&A] [answered]
Hello members,
2 globe views, one of Mapbox and other of deck.gl are incompatible with each other.  Radius is different for 2 spheres.
Even if radius is corrected, then also, enabling pitch would be some extra work.

Anyone has more ideas about this? any plans for this feature?

<img width="668" alt="image" src="https://user-images.githubusercontent.com/10899081/177044208-82c060a8-7687-406a-a1b3-25e114caf0bc.png">

**Comment by Animus-Development** [comment]
@rajatmann100 - I'm coming to this late but have been keen to explore a solution for some test vis. 

In my case I'm using, React with something like this: 
`
<DeckGL
      ref={deckRef}
      layers={[layers]}
      views={[globeView]}
    >
      <Map
        ref={mapRef}
        projection="globe"
        mapStyle={`mapbox://styles/${mapStyle}`}
        mapboxAccessToken={process.env.ACCESS_TOKEN}
      />
    </DeckGL>
`

Have a look at the mapbox-gl module (/mapbox-gl/dist/mapbox-gl-unminified.js) line 51698. (in package.json of the module change the "main": to "dist/mapbox-gl-unminified.js").

`
    _worldSizeFromZoom(zoom) {
        return Math.pow(2, zoom) * this.tileSize;
    }
`

You can influence the size of the mapbox globe by adding something like `return Math.pow(2, zoom) * this.tileSize * {something};`

I'd encourage you (where possible) to explore a less heavy handed solution.

**Comment by chrisgervang** [comment, marked as answer]
Hey there! `GlobeView` is supported by Maplibre. https://deck.gl/docs/whats-new#globe-view--maplibre

The dynamic flat to globe transition in Mapbox cannot be integrated with deck.gl as they have not opened up an camera state API. Mapbox is closed source now, so we don't have the ability to freely collaborate anymore. If it's important to you, please ask them to open their camera API up!
